### ISSUE TITLE
string literal macros

### ISSUE THREAD

**Issue by FuZhiyu** [matthieugomez/MonthlyDates.jl#13]
I got lazy again and found the construction of a single quarterly date is a bit tedious (`QuarterlyDate(2020, 1)`). Do you consider adding string literal macros to the package? Here is my idea:

```julia
macro yq_str(ymstr)
    return :(QuarterlyDate($ymstr))
end

macro ym_str(ymstr)
    return :(MonthlyDate($ymstr))
end
```
so a `QuarterlyDate(2020, 3)` can be simply constructed as `yq"2020-Q3"`.

**Comment by matthieugomez** [comment]
Maybe. Do these types of string literal macros to construct dates exist anywhere else?

**Comment by matthieugomez** [comment]
Note that you can already do `QuarterlyDate("2020-Q3")`

**Comment by FuZhiyu** [comment]
> Maybe. Do these types of string literal macros to construct dates exist anywhere else?

No. But I guess that's because `Date` is relatively easy to type while `QuarterlyDate` is pretty tedious. 

Your call. I can also simply put it into my own header file.
